samples/drivers/led_lp5569/src/main.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2023 Phytec Messtechnik GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/led.h>
+#include <zephyr/kernel.h>
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(app);
+
+#define NUM_LEDS 9
+#define DELAY_TIME	K_MSEC(1000)
+
+int main(void)
+{
+	const struct device *const led_dev = DEVICE_DT_GET_ANY(ti_lp5569);
+	int i, ret;
+
+	if (!led_dev) {
+		LOG_ERR("No device with compatible ti,lp5569 found");
+		return 0;
+	} else if (!device_is_ready(led_dev)) {
+		LOG_ERR("LED device %s not ready", led_dev->name);
+		return 0;
+	}
+
+	LOG_INF("Found LED device %s", led_dev->name);
+
+	/*
+	 * Display a continuous pattern that turns on 9 LEDs at 1 s one by
+	 * one until it reaches the end and turns off LEDs in reverse order.
+	 */
+
+	LOG_INF("Testing 9 LEDs ..");
+
+	while (1) {
+		/* Turn on LEDs one by one */
+		for (i = 0; i < NUM_LEDS; i++) {
+			ret = led_on(led_dev, i);
+			if (ret < 0) {
+				return 0;
+			}
+
+			k_sleep(DELAY_TIME);
+		}
+
+		/* Turn off LEDs one by one */
+		for (i = NUM_LEDS - 1; i >= 0; i--) {
+			ret = led_off(led_dev, i);
+			if (ret < 0) {
+				return 0;
+			}
+
+			k_sleep(DELAY_TIME);
+		}
```

Comment:
Please can you add a breathing test as well ? To demonstrate the `set_brightness` API function.

Reply:
Good point! I changed the sample so that it enables all leds step-by-step and then slowly dims them simultaneously until off.